RetakesPlugin/Modules/Managers/GameManager.cs

```diff
@@ -66,40 +69,34 @@ public void AddScore(CCSPlayerController player, int score)
 
     private int _consecutiveRoundsWon;
 
-    public void TerroristRoundWin()
+    private void TerroristRoundWin()
     {
         _consecutiveRoundsWon++;
 
-        if (_consecutiveRoundsWon == _consecutiveRoundWinsToScramble)
+        bool shouldScrambleNow = _isScrambleEnabled && _consecutiveRoundsWon == _consecutiveRoundWinsToScramble;
+        int roundsLeftToScramble = _consecutiveRoundWinsToScramble - _consecutiveRoundsWon;
+        // Almost scramble if 1-2 rounds left to automatic scramble
+        bool shouldAlmostScramble = _isScrambleEnabled && 0 < roundsLeftToScramble && roundsLeftToScramble <= 2;
+
+        if (shouldScrambleNow)
         {
             Server.PrintToChatAll(
                 $"{RetakesPlugin.MessagePrefix}{_translator["retakes.teams.scramble", _consecutiveRoundWinsToScramble]}");
 
-            _consecutiveRoundsWon = 0;
             ScrambleTeams();
         }
-        else if (_consecutiveRoundsWon >= 3)
-        {
-            if (_isScrambleEnabled)
-            {
-                Server.PrintToChatAll(
-                    $"{RetakesPlugin.MessagePrefix}{_translator["retakes.teams.almost_scramble", _consecutiveRoundsWon, _consecutiveRoundWinsToScramble - _consecutiveRoundsWon]}");
-            }
-            else
-            {
-                Server.PrintToChatAll(
-                    $"{RetakesPlugin.MessagePrefix}{_translator["retakes.teams.win_streak", _consecutiveRoundsWon]}");
-            }
+        else if (shouldAlmostScramble) {
```

Comment:
Please make sure to follow the existing project formatting

Reply:
Will fix, I forgot to run format on save